analyzers/tests/SonarAnalyzer.UnitTest/Rules/UnusedStringBuilderTest.cs

```diff
@@ -0,0 +1,174 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2023 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using CS = SonarAnalyzer.Rules.CSharp;
+using VB = SonarAnalyzer.Rules.VisualBasic;
+
+namespace SonarAnalyzer.UnitTest.Rules;
+
+[TestClass]
+public class UnusedStringBuilderTest
+{
+    private readonly VerifierBuilder builderCS = new VerifierBuilder<CS.UnusedStringBuilder>();
+    private readonly VerifierBuilder builderVB = new VerifierBuilder<VB.UnusedStringBuilder>();
+
+    [TestMethod]
+    public void UnusedStringBuilder_CS() =>
+        builderCS.AddPaths("UnusedStringBuilder.cs").Verify();
+
+    [TestMethod]
+    public void UnusedStringBuilder_VB() =>
+        builderVB.AddPaths("UnusedStringBuilder.vb").Verify();
+
+#if NET
+
+    [TestMethod]
+    public void UnusedStringBuilder_CSharp9() =>
+        builderCS.AddPaths("UnusedStringBuilder.CSharp9.cs")
+            .WithOptions(ParseOptionsHelper.FromCSharp9)
+            .Verify();
+
+    [DataTestMethod]
+    [DataRow("", false)]
+    [DataRow("sb.ToString();", true)]
+    [DataRow("""var a = sb.Append("").Append("").Append("").Append("").ToString().ToLower();""", true)]
+    [DataRow("sb.CopyTo(0, new char[1], 0, 1);", true)]
+    [DataRow("sb.GetChunks();", true)]
+    [DataRow("var a = sb[0];", true)]
+    [DataRow("""sb?.Append("").ToString().ToLower();""", false)] // FP
```

Comment:
https://github.com/SonarSource/sonar-dotnet/issues/6743

Reply:
You can add the issue number to the FP
```suggestion
    [DataRow("""sb?.Append("").ToString().ToLower();""", false)] // FP see https://github.com/SonarSource/sonar-dotnet/issues/6743
```